wordpress-mobile/WordPress-Android
# Issue/10712 material quick start

Fixes #10712 

This PR moves Quick Start and Promo Dialog screens to Material Theme and adds Dark Theme support. There are some code changes for cosmetic reasons. 

Quick Start card on My Site fragment is a part of another review.

Design changes are reviewed before the code review is requested.

While refactoring is not in the scope of this PR, since I touched those files, I cleaned up/optimized some of the redundant code as suggested by AS, so do not be alarmed by the size of the PR. I suggest turning on the "Hide whitespace changes" option during the review.

There is a huge amount of PRs in this feature that covers all the screens within the app, so I'm requesting reviews from a wide variety of people.

![comparison_quickstart](https://user-images.githubusercontent.com/728822/72653858-cd67a280-3941-11ea-88b5-5669fe8aa6be.png)


To test:
- Create a new Site, and confirm the visual integrity of Quick Start promo dialog with both Light and Dark themes.
- Navigate to Quick Start screen and confirm it's visual integrity with both Light and Dark themes.
 The theme can be changed from App Settings.



PR submission checklist:
